lib/Remote/RemoteTargetLocator.php

```diff
@@ -68,10 +68,22 @@ public function defaultContent()
      */
     public function frame($frame)
     {
-        if ($frame instanceof WebDriverElement) {
-            $id = ['ELEMENT' => $frame->getID()];
+        if ($this->isW3cCompliant) {
+            if ($frame instanceof WebDriverElement) {
+                $id = [JsonWireCompat::WEB_DRIVER_ELEMENT_IDENTIFIER => $frame->getID()];
+            } else if ($frame === null) {
+                $id = null;
+            } else if (is_numeric($frame)) {
+                $id = (int) $frame;
+            } else {
+                $id = (string) $frame;
```

Comment:
If W3C only allows $frame to be either `WebDriverElement`, `null` or numeric ID, why should we accept string (and send it to the driver)?

Reply:
So do you think this should except? The only reason I didn’t do so was because I wasn’t sure what exception I should throw. I haven’t seen any/many examples where we throw an exception rather than translate an error into an exception.

Should we just throw the same exception that would be thrown anyway?

Reply:
We can just throw `\InvalidArgumentException` (the PHP's InvalidArgumentException, not the one from WebDriver). We already do it elsewhere in similar case.